lhafile/lhafile.py

```diff
@@ -306,6 +307,25 @@ def _RealGetContent(self):
             directory = directory.decode("ISO-8859-1")
             directory = os.sep.join(directory.split('\xff'))
             filename = os.path.join(directory, filename)
+
+        # OSIDs based on
+        # http://dangan.g.dgdg.jp/en/Content/Program/Java/jLHA/Notes/Notes.html
+        os_identifiers = { 0x41: 'Amiga', 0x4D: 'MS-DOS', 0x32: 'OS/2', 0x39: 'OS-9',
+                          0x4B: 'OS/68K', 0x33: 'OS/386', 0x48: 'Human68K', 0x55: 'UNIX',
+                          0x43: 'CP/M', 0x46: 'FLEX', 0x6D: 'Macintosh', 0x52: 'Runser',
+                          0x54: 'TownOS', 0x58: 'XOSK', 0x77: 'Windows95', 0x57: 'WindowsNT',
+                          0x00: 'MS-DOS', 0x4A: 'Java' }
+        create_system = os_identifiers.get(os_identifier)

```

Comment:
Probably should add a default value where so it does not fail when OS identifier is not recognized, e.g.:

    create_system = os_identifiers.get(os_identifier, "")

Reply:
Hm, it wouldn't really fail. It would be set to *None*, which may be considered wrong because in fact some system **has** created the archive. :-) I let it set to an empty string as you suggested.